storybookjs/storybook
# Maintenance: Remove need for `react` as peerDependency

Reverts https://github.com/storybookjs/storybook/pull/23882



## What I did

I found out the problem that was causing https://github.com/storybookjs/storybook/issues/23876, so instead of reverting the whole thing, we fix it proper.

## How to test

To reproduce:

1. clone the [react-parallax-tilt](https://github.com/mkosir/react-parallax-tilt) repo
2. checkout the `v1.7.161` tag, which is where the issue was found: https://github.com/mkosir/react-parallax-tilt/tree/v1.7.161
4. run `npm run start-storybook` - it should work
5. remove `@storybook/addon-essentials` and `@storybook/manager-api`
6. run `npm install` then `npm run start-storybook` - it should fail
7. still with those dependencies removed, change the storybook dependencies to the canary of this PR: `0.0.0-pr-23897-sha-78c2d886`
8. run `npm run start-storybook` - it should work!

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers


`["cleanup", "BREAKING CHANGE", "feature request", "bug", "build", "documentation", "maintenance", "dependencies", "other"]`



### 🦋 Canary release


This pull request has been released as version [`0.0.0-pr-23897-sha-78c2d886`](https://npmjs.com/package/@storybook/cli/v/0.0.0-pr-23897-sha-78c2d886). Install it by pinning all your Storybook dependencies to that version.
<details>
<summary>More information</summary>

| | |
| --- | --- |
| **Published version** | [`0.0.0-pr-23897-sha-78c2d886`](https://npmjs.com/package/@storybook/cli/v/0.0.0-pr-23897-sha-78c2d886) |
| **Triggered by** | @yannbf |
| **Repository** | [storybookjs/storybook](https://github.com/storybookjs/storybook) |
| **Branch** | [`revert-23882-revert-23496-norbert/react-peerdep-webpack-builder-removal`](https://github.com/storybookjs/storybook/tree/revert-23882-revert-23496-norbert/react-peerdep-webpack-builder-removal) |
| **Commit** | [`78c2d886`](https://github.com/storybookjs/storybook/commit/78c2d88695e7fd1b4cc91b7c7c3136e0da2a506f) |
| **Datetime** | Mon Aug 21 07:26:27 UTC 2023 (`1692602787`) |
| **Workflow run** | [5923278529](https://github.com/storybookjs/storybook/actions/runs/5923278529) |

To request a new release of this pull request, mention the `@storybookjs/core` team.

_core team members can create a new canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=23897`_
</details>
